packages/transformer/src/test/standalone/BranchProvenanceInitializer.test.ts

```diff
@@ -0,0 +1,287 @@
+import * as fs from "fs";
+import { ElementGroupsMembers, ExternalSource, ExternalSourceAspect, ExternalSourceIsInRepository, IModelDb, IModelHost, PhysicalModel, PhysicalObject, RepositoryLink, SpatialCategory, StandaloneDb } from "@itwin/core-backend";
+import { ProvenanceInitArgs, ProvenanceInitResult, initializeBranchProvenance } from "../../BranchProvenanceInitializer";
+import { IModelTransformerTestUtils, assertIdentityTransformation } from "../IModelTransformerUtils";
+import { BriefcaseIdValue, Code } from "@itwin/core-common";
+import { IModelTransformer } from "../../IModelTransformer";
+import { Guid, OpenMode, TupleKeyedMap } from "@itwin/core-bentley";
+import { expect } from "chai";
+import { Point3d, YawPitchRollAngles } from "@itwin/core-geometry";
+
+describe("compare imodels from BranchProvenanceInitializer and traditional branch init", () => {
+  // truth table (sourceHasFedGuid, targetHasFedGuid, forceCreateFedGuidsForMaster) -> (relSourceAspectNum, relTargetAspectNum)
+  const sourceTargetFedGuidToAspectCountMap = new TupleKeyedMap([
+    [[false, false, false], [2, 1]],
+    // "keep-reopened-db" is truthy but also equal to an optimized optional argument that we use
+    [[false, false, "keep-reopened-db"], [0, 0]],
+    [[false, true, false], [2, 0]],
+    [[false, true, "keep-reopened-db"], [0, 0]],
+    [[true, false, false], [1, 1]],
+    [[true, false, "keep-reopened-db"], [0, 0]],
+    [[true, true, false], [0, 0]],
+    [[true, true, "keep-reopened-db"], [0, 0]],
+  ]);
+
+  let index = 0;
+  // FIXME: don't use a separate iModel for each loop iteration, just add more element pairs
+  // to the one iModel. That will be much faster
+  for (const sourceHasFedguid of [true, false]) {
+    for (const targetHasFedguid of [true, false]) {
+      for (const createFedGuidsForMaster of ["keep-reopened-db", false] as const) {
+        it(`branch provenance init with ${[
+          sourceHasFedguid && "relSourceHasFedGuid",
+          targetHasFedguid && "relTargetHasFedGuid",
+          createFedGuidsForMaster && "createFedGuidsForMaster",
+        ].filter(Boolean)
+         .join(",")
+        }`, async () => {
+          let transformerMasterDb!: StandaloneDb;
+          let noTransformerMasterDb!: StandaloneDb;
+          let transformerForkDb!: StandaloneDb;
+          let noTransformerForkDb!: StandaloneDb;
+
+          try {
+            const suffixName = (s: string) => `${s}_${sourceHasFedguid ? "S" : "_"}${targetHasFedguid ? "T" : "_"}${createFedGuidsForMaster ? "C" : "_"}.bim`;
+            const sourceFileName = suffixName("ProvInitSource");
+            // const generatedIModel = generateIModel(sourceFileName, sourceHasFedguid, targetHasFedguid);
```

Comment:
you aren't using this utility... either delete it or go back to using it and move the setup logic to it. I could go either way on inline or separate.